nfc/libs/NfcCoreLib/lib/Nci/phNciNfc_Init.c

```diff
@@ -145,31 +347,132 @@ static NFCSTATUS phNciNfc_ProcessInitRsp(void *pContext, NFCSTATUS Status)
         {
             wStatus = NFCSTATUS_FAILED;
         }
+    }
+    else
+    {
+        wStatus = NFCSTATUS_INVALID_PARAMETER;
+    }
+    PH_LOG_NCI_FUNC_EXIT();
+    return wStatus;
+}
+
+static void phNciNfc_DelayForCreditNtfCb(void* pContext, uint8_t bCredits, NFCSTATUS status)
+{
+    pphNciNfc_Context_t pNciContext = (pphNciNfc_Context_t)pContext;
+    phNciNfc_TransactInfo_t tTranscInfo;
+    NFCSTATUS wStatus = NFCSTATUS_SUCCESS;
+
+    PH_LOG_NCI_FUNC_ENTRY();
+
+    if ((NFCSTATUS_SUCCESS != status) || (0 == bCredits))
+    {
+        /* setting generic WaitCreditTimeout event to enable processing of state machine even in failure case */
+        wStatus = NFCSTATUS_CREDIT_TIMEOUT;
+        PH_LOG_NCI_CRIT_STR("Credits Update from NciNfc_Init Module Failed");
+        phNciNfc_Notify(pNciContext, wStatus, NULL);
+    }
+    else
+    {
+        if (NULL != pNciContext && PH_NCINFC_VERSION_IS_2x(pNciContext))
+        {
+            tTranscInfo.pContext = (void*)pNciContext;
+            tTranscInfo.pbuffer = (void*)&pNciContext->ResetInfo.ResetTypeRsp;
+            tTranscInfo.wLength = sizeof(pNciContext->ResetInfo.ResetTypeRsp);
+            phNciNfc_Notify(pNciContext, wStatus, (void *)&tTranscInfo);
+        }
+        else
+        {
+            wStatus = NFCSTATUS_INVALID_STATE;
+            phNciNfc_Notify(pNciContext, wStatus, NULL);
+        }
+    }
+
+    PH_LOG_NCI_FUNC_EXIT();
+}
+
+static void phNciNfc_ResetNtfDelayCb(uint32_t dwTimerId, void *pContext)
+{
+    pphNciNfc_Context_t pNciContext = (pphNciNfc_Context_t)pContext;
+    /* Internal event is made failed in order to remain in same state */
+    NFCSTATUS wStatus = NFCSTATUS_SUCCESS;
+    UNUSED(dwTimerId);
+    PH_LOG_LIBNFC_FUNC_ENTRY();
+    if (NULL != pNciContext)
+    {
+        (void)phOsalNfc_Timer_Stop(pNciContext->dwNtfTimerId);
+        (void)phOsalNfc_Timer_Delete(pNciContext->dwNtfTimerId);
+
+        pNciContext->dwNtfTimerId = 0;
 
-        if((NFCSTATUS_SUCCESS == wStatus) &&
-            (0 == pNciContext->tInitInfo.bSkipRegisterAllNtfs))
+        (void)phNciNfc_GenericSequence(pNciContext, NULL, wStatus);
```

Comment:
On phNciNfc_GenericSequence() failure just leave an error message with failure status in the logs - it's enough for here. For timer functions might consider warning messages - main point is not to ignore failures, they can be very hard to identify if impact is not straight-forward